runtime/src/iree/tooling/device_util.c

```diff
@@ -106,12 +106,18 @@ static iree_status_t iree_hal_flags_list_driver_devices(
     iree_hal_driver_registry_t* driver_registry, iree_string_view_t driver_name,
     iree_allocator_t host_allocator, FILE* file) {
   iree_hal_driver_t* driver = NULL;
-  IREE_RETURN_IF_ERROR(iree_hal_driver_registry_try_create(
-      driver_registry, driver_name, host_allocator, &driver));
+  iree_status_t status = iree_hal_driver_registry_try_create(
+      driver_registry, driver_name, host_allocator, &driver);
+  if (!iree_status_is_ok(status)) {
+    // Driver could not be created (not found, unavailable, etc) and so there
+    // are no devices to list.
+    iree_status_ignore(status);
+    return iree_ok_status();
```

Comment:
How about matching what the CTS does - looking for specific error types?
https://github.com/iree-org/iree/blob/065fe79c7bee919d287aa3cf156e821fb42044db/runtime/src/iree/hal/cts/cts_test_base.h#L45-L56

(that gets fuzzy pretty quickly though, since drivers _can_ return other error types)

Reply:
Hmm maybe - the command here is intended to be something that can be passed to xargs and can't have any extra output so there's not much I'd want to do besides ignore it or die, and dying feels wrong for a "list me available for use" command as being unavailable for any reason just means it's not available for use and shouldn't be in the list. Could possibly change behavior based on whether doing `--list_devices` or `--list_devices=cuda`, but eh. Will wait to see if there's more confusion :)

Reply:
Yeah, we could differentiate between "these are all the devices the program _could try to_ create" and "these are all the devices the program _can successfully_ create"